agent/consul/leader_metrics.go

```diff
@@ -0,0 +1,71 @@
+package consul
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/armon/go-metrics/prometheus"
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/consul/logging"
+)
+
+var CertExpirationGauges = []prometheus.GaugeDefinition{
+	{
+		Name: metricsKeyMeshRootCAExpiry,
+		Help: "Seconds until the service mesh root certificate expires.",
+	},
+}
+
+var metricsKeyMeshRootCAExpiry = []string{"mesh", "root-ca", "expiry"}
+
+func rootCAExpiryMonitor(s *Server) certExpirationMonitor {
+	return certExpirationMonitor{
+		Key: metricsKeyMeshRootCAExpiry,
+		Labels: []metrics.Label{
+			{Name: "datacenter", Value: s.config.Datacenter},
+		},
+		Logger: s.logger.Named(logging.Connect),
+		Query: func() (time.Duration, error) {
+			state := s.fsm.State()
+			_, root, err := state.CARootActive(nil)
```

Comment:
Could the root be nil?

Reply:
I guess it could be nil. I added a case to handle that to prevent the panic.